src/osp/Shaders/Flat.cpp

```diff
@@ -89,12 +86,12 @@ void shader::assign_flat(
             if (diffuse.contains(ent))
             {
                 pStorageOpaque->emplace(
-                        ent, EntityToDraw{&draw_ent_flat, {&rData, &(*rData.m_shaderDiffuse)} });
+                        ent, EntityToDraw{&draw_ent_flat, {&rData, &rData.m_shaderDiffuse} });
```

Comment:
consider std::addressof() for more genericness.

Reply:
Will this function benefit from the genericness? This just seems like extra clutter that makes it more difficult to read in this case.